File: pt_app/models/ml_info.py
import time
from sqlalchemy import (Column, Integer, String,
                        Boolean, BigInteger, ForeignKey,
                        DateTime, Float, UUID, func, ARRAY)
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import relationship
from sqlalchemy import *
from sqlalchemy.orm import *
from models.utils import BaseModel, Base


class ModelInfo(BaseModel):
    name: str
    path: str
    metrics: str
    accuracy: float
    feature_importance: dict
    created: DateTime

    __tablename__ = "model_infos"

    id = Column(Integer, primary_key=True, autoincrement=True)

    name = Column(String, nullable=True)
    path = Column(String, nullable=True)
    metrics = Column(String, nullable=True)
    accuracy = Column(Float, nullable=True, default=False)
    feature_importance = Column(JSONB, nullable=True, default='{}')
    created = Column(DateTime(timezone=True), nullable=True)

File: go_app/pkg/models/ml_info.go
package models

import (
	"encoding/json"
	"time"
)

type ModelInfo struct {
	ID                uint64     `gorm:"primaryKey" json:"id"`
	Name              string     `gorm:"unique" json:"name"`
	Metrics           string     `json:"metrics"`
	Path              string     `json:"path"`
	Accuracy          float64    `json:"accuracy"`
	FeatureImportance FeatureImp `gorm:"type:jsonb;default:'{}';" json:"feature_importance"`
	Created           time.Time  `gorm:"autoUpdateTime" json:"created"`
}

type FeatureImp struct {
	FeatureImportances []FeatureImportance `json:"feature_importances"`
}

type FeatureImportance struct {
	Name  string  `json:"name"`
	Score float64 `json:"score"`
}

func (s *FeatureImp) Scan(v interface{}) error {
	err := json.Unmarshal(v.([]byte), &s)
	if err != nil {
		return err
	}
	return nil
}
